src/httprpc.cpp

```diff
@@ -307,9 +304,11 @@ static bool InitRPCAuthentication()
             cookie_perms = *perm_opt;
         }
 
+        assert(strRPCUserColonPass.empty()); // Only support initializing once
         if (!GenerateAuthCookie(&strRPCUserColonPass, cookie_perms)) {
             return false;
         }
+        if (!strRPCUserColonPass.empty()) LogInfo("Using random cookie authentication.");
```

Comment:
what if it's empty?

Reply:
Moved back the log message in latest push to clarify.